addons/mail/static/src/components/message/message.js

```diff
@@ -126,24 +126,19 @@ class Message extends Component {
     // Public
     //--------------------------------------------------------------------------
 
+    /**
+     * @returns {string}
+     */
+    get anonymousName() {
+        return (this.message.originThread && this.message.originThread.anonymous_name) || this.env._t("Anonymous");
+    }
+
     /**
      * @returns {string}
      */
     get avatar() {
-        if (
-            this.message.author &&
-            this.message.author === this.env.messaging.partnerRoot
-        ) {
-            return '/mail/static/src/img/odoobot.png';
-        } else if (this.message.author) {
-            // TODO FIXME for public user this might not be accessible. task-2223236
-            // we should probably use the correspondig attachment id + access token
-            // or create a dedicated route to get message image, checking the access right of the message
-            return `/web/image/res.partner/${this.message.author.id}/image_128`;
-        } else if (this.message.message_type === 'email') {
-            return '/mail/static/src/img/email_icon.png';
-        }
-        return '/mail/static/src/img/smiley/avatar.jpg';
+        // use private method, so it can be overridden
+        return this._getAvatar();
```

Comment:
I would consider moving this to the message js model as a computed field.

Reply:
@seb-odoo It will break the existing code (as it was already used)

Reply:
What do you mean? You can update existing code if necessary to call/override the method from the model instead of from the component.